### PR TITLE
Fix nested fqn discovery

### PR DESCRIPTION
**Pull request by albanD**
I think I missed some fix!

### CODE REVIEW

File: torch/utils/module_tracker.py [hunk 1]
```diff
@@ -81,6 +81,8 @@ def _get_mod_name(self, mod):
         if mod not in self._seen_modules:
             for name, submod in mod.named_children():
                 self._known_modules[submod] = f"{mod_name}.{name}"
+                self._get_mod_name(submod)
```
Comment (sanketpurandare) [inline torch/utils/module_tracker.py:84]: Neat solution!

File: test/test_module_tracker.py [hunk 2]
```diff
@@ -25,10 +25,10 @@ class Mod(torch.nn.Module):
             def __init__(self):
                 super().__init__()
                 self.a = Foo()
-                self.b = Foo()
+                self.b = torch.nn.ModuleDict({"nest": Foo()})
```
Comment (sanketpurandare) [inline test/test_module_tracker.py:28]: Should we also just add a test for nn.ModuleList just to be safe?

Comment (albanD) [reply test/test_module_tracker.py:28]: Added

**Review by sanketpurandare — APPROVED** [review]

**Review by janeyx99 — APPROVED** [review]

**Pull request closed without merging**